generators/upgrade/index.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+var util = require('util'),
+    generators = require('yeoman-generator'),
+    chalk = require('chalk'),
+    scriptBase = require('../generator-base'),
+    packagejs = require('../../package.json'),
+    fs = require('fs'),
+    shelljs = require('shelljs'),
+    exec = require('child_process').exec,
+    semver = require('semver'),
+    Git = require('git-wrapper');
+
+var UpgradeGenerator = generators.Base.extend({});
+
+util.inherits(UpgradeGenerator, scriptBase);
+
+/* Constants used throughout */
+const GENERATOR_JHIPSTER = 'generator-jhipster',
+    UPGRADE_BRANCH = 'jhipster_upgrade';
+
+module.exports = UpgradeGenerator.extend({
+    constructor: function () {
+        generators.Base.apply(this, arguments);
+
+        this.currentVersion = packagejs.version;
+        this.force = true;
+        this.git = new Git();
+
+        this.gitCheckout = function(branch) {
+            var done = this.async();
+            this.git.exec('checkout', {q: true}, [branch], function(err, msg) {
+                if (err != null) throw new Error(err);
+                this.log('Checked out branch ' + branch);
+                done();
+            }.bind(this));
+        };
+    },
+
+    initializing: {
+        displayLogo: function () {
+            this.printJHipsterLogo();
+        }
+    },
+
+    configuring: {
+        assertGitPresent: function() {
+            var done = this.async();
+            exec('git --version', function (err) {
+                if (err) {
+                    this.log(chalk.yellow.bold('WARNING!') + ' git is not found on your computer.\n',
+                        ' Install git: ' + chalk.yellow('http://git-scm.com/')
```

Comment:
Yeoman has an inbuit error method. Could you check our sub generator where its used. Eg the entity generator when validations fail